server/api/leaderboard.js

```diff
@@ -0,0 +1,67 @@
+const db = require('../database')
+
+const { responses } = require('../responses')
+const challenges = require('../challenges')
+const { getScore } = require('../util/scores')
+
+module.exports = {
+  method: 'get',
+  path: '/leaderboard',
+  requireAuth: false,
+  schema: {},
+  handler: async ({ req, uuid }) => {
+
+    let solveAmount = {}
+    let challengeValues = {}
+    let userSolves = {}
+    let userScores = []
+    
+    const solves = await db.leaderboard.getSolves()
+    const users = await db.leaderboard.getUsers()
+
+    for(let i = 0; i < solves.length; i++){
+      // Accumulate in solveAmount
+      if(!(solves[i].challengeid in solveAmount)){
+        solveAmount[solves[i].challengeid] = 1
+      }else{
+        solveAmount[solves[i].challengeid] += 1
+      }
+      // Store which challenges each user solved for later
+      if(!(solves[i].userid in userSolves)){
+        userSolves[solves[i].userid] = [solves[i].challengeid]
+      }else{
+        userSolves[solves[i].userid].push(solves[i].challengeid)
+      }
+    }
+
+    for(let i = 0; i < challenges.getAllChallenges().length; i++){
```

Comment:
`challenges.getAllChallenges()` should be stored in a local variable? It might be expensive to call